Lib/logging/handlers.py

```diff
@@ -1015,8 +1015,22 @@ def emit(self, record):
                     self.socket.close()
                     self._connect_unixsocket(self.address)
                     self.socket.send(msg)
+            # UDP socket type 
             elif self.socktype == socket.SOCK_DGRAM:
                 self.socket.sendto(msg, self.address)
+            # TCP socket type
+            # This block doesn't exist in original handlers.py code for TCP 
+            elif self.socktype == socket.SOCK_STREAM:
+                try:
+
+                    self.socket.sendall(msg)
+                # create new TCP socket stream, if TCP socket broken or disconnected 
+                except OSError:
+
```

Comment:
Do this for _any_ error? Wouldn't just overwriting the `self.socket` leak the resources of the existing socket, at least in some circumstances?